### PR TITLE
[Feature] Support ade20k dataset

### PR DESCRIPTION
**Pull request by xiexinch**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMCls.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: tools/misc/download_dataset.py [hunk 1]
```diff
@@ -176,6 +176,16 @@ def main():
             'https://dorc.ks3-cn-beijing.ksyun.com/data-set/2020Objects365%E6%95%B0%E6%8D%AE%E9%9B%86/val/images/v1/',  # noqa
             # validation url root_2
             'https://dorc.ks3-cn-beijing.ksyun.com/data-set/2020Objects365%E6%95%B0%E6%8D%AE%E9%9B%86/val/images/v2/'  # noqa
+        ],
+        ade20k=[
```
Comment (hhaAndroid) [inline tools/misc/download_dataset.py:180]: ade20k 有多个版本吗？如果有最后加一些后缀，例如 ade20k-2016 这种

File: tools/dataset_converters/ade20k2coco.py [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+from panopticapi.utils import IdGenerator, save_json
+from PIL import Image
+
+from mmdet.datasets.ade20k import ADE20KPanopticDataset
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description='Convert ADE20K annotations to COCO format')
+    parser.add_argument('ade20k_path', help='ade20k data path')
```
Comment (hhaAndroid) [inline tools/dataset_converters/ade20k2coco.py:15]:
```suggestion
    parser.add_argument('src', help='ade20k data path')
```

Comment (hhaAndroid) [reply tools/dataset_converters/ade20k2coco.py:15]: 因为这个是inplace 的计算，需要在程序运行前一开始就print告诉用户，我们会在 src 文件夹下新增 ann 文件夹。如果 ann 文件夹已经存在我觉得最好直接报错，不然覆盖掉用户文件可能会被喷

File: configs/_base_/datasets/ade20k_panoptic.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+backend_args = None
+
+train_pipeline = [
+    dict(type='LoadImageFromFile', backend_args=backend_args),
+    dict(type='LoadPanopticAnnotations', backend_args=backend_args),
+    dict(
+        type='RandomChoiceResize',
+        scales=[int(512 * x * 0.1) for x in range(5, 21)],
+        resize_type='ResizeShortestEdge',
```
Comment (hhaAndroid) [inline configs/_base_/datasets/ade20k_panoptic.py:13]: 换成 FixScaleResize，然后加个 TODO 说明暂时未验证

File: mmdet/datasets/transforms/transforms.py [hunk 4]
```diff
@@ -414,6 +414,90 @@ def __repr__(self) -> str:
         return repr_str
 
 
+@TRANSFORMS.register_module()
+class ResizeShortestEdge(BaseTransform):
```
Comment (hhaAndroid) [inline mmdet/datasets/transforms/transforms.py:418]: 这个先删掉吧

File: configs/_base_/datasets/ade20k_panoptic.py [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+dataset_type = 'ADE20KPanopticDataset'
+data_root = 'data/ADEChallengeData2016/'
+
+backend_args = None
+
+train_pipeline = [
+    dict(type='LoadImageFromFile', backend_args=backend_args),
+    dict(type='LoadPanopticAnnotations', backend_args=backend_args),
+    # TODO: the performance of `FixScaleResize` need to check.
+    dict(type='FixScaleResize', scale=(2560, 640), backend_args=backend_args),
```
Comment (hhaAndroid) [inline configs/_base_/datasets/ade20k_panoptic.py:11]: 2560,640 是正确的配置吗？

Comment (xiexinch) [reply configs/_base_/datasets/ade20k_panoptic.py:11]: https://github.com/facebookresearch/Mask2Former/blob/main/configs/ade20k/panoptic-segmentation/Base-ADE20K-PanopticSegmentation.yaml#L37
mask2former是这样配置的

**Review by hhaAndroid — APPROVED** [review]

**Pull request merged**